packages/myst-transforms/src/toc.ts

```diff
@@ -22,38 +30,52 @@ function listFromPages(pages: ProjectPage[], projectSlug?: string): List {
       if (page.level !== level) return undefined;
       return listItemFromPages(pages.slice(index), projectSlug);
     })
+    // Remove any null items
     .filter((item): item is ListItem => !!item);
-  return { type: 'list', children };
+  return makeList(children);
 }
 
-function listItemFromPages(pages: ProjectPage[], projectSlug?: string) {
+function listItemFromPages(pages: ProjectPage[], projectSlug?: string): ListItem {
   if (pages.length === 0) return;
-  const { title, slug, enumerator, level } = pages[0];
+  const page = pages[0];
+  const child = listItemChildFromPage(page, projectSlug);
+  const item = makeListItem(child);
+  if (pages[1] && pages[1].level > page.level) {
+    item.children.push(listFromPages(pages.slice(1), projectSlug));
+  }
+  return item;
+}
+
+function listItemChildFromPage(page: ProjectPage, projectSlug?: string): Text | Link {
```

Comment:
Confusing function name they used here, took me a while to figure out what's going on :upside_down_face: